test/Microsoft.VisualStudio.Editor.Razor.Test/Infrastructure/TestTextBuffer.cs

```diff
@@ -49,12 +56,27 @@ public void ApplyEdits(params TestEdit[] edits)
             ContentTypeChanged?.Invoke(null, null);
         }
 
+        public IReadOnlyList<EventHandler<TextContentChangedEventArgs>> AttachedChangedEvents => _attachedChangedEvents;
+
         public ITextSnapshot CurrentSnapshot => _currentSnapshot;
 
         public PropertyCollection Properties { get; }
 
         public event EventHandler<SnapshotSpanEventArgs> ReadOnlyRegionsChanged;
-        public event EventHandler<TextContentChangedEventArgs> Changed;
+
+        public event EventHandler<TextContentChangedEventArgs> Changed
+        {
+            add
+            {
+                _attachedChangedEvents.Add(value);
+            }
+            remove
+            {
+                _attachedChangedEvents.Remove(value);
+            }
+        }
```

Comment:
lol why

Reply:
"The Taylor" 2.0

Reply:
Unit testability. We have several classes that attach/detach to buffers based on criteria happening. It'd be awful to do an `Assert.Throws(InvokeChangedEvent)` to determine of a class has attached to the buffer.